tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -310,52 +310,48 @@ private async Task UploadLogLinesBlobAsync(string account, Build build, BuildLog
                     return;
                 }
 
-                using var tempFile = new DisposableTempFile();
+                var messagesWriter = new StringBuilder();
+                var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
+                var lastTimeStamp = log.CreatedOn;
 
-                await using (var messagesWriter = new StreamWriter(File.OpenWrite(tempFile.Path)))
+                for (var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
                 {
-                    var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
-                    var lastTimeStamp = log.CreatedOn;
+                    var line = logLines[lineNumber - 1];
+                    var match = Regex.Match(line, @"^([^Z]{20,28}Z) (.*)$");
+                    var timestamp = match.Success
+                        ? DateTime.ParseExact(match.Groups[1].Value, TimeFormat, null,
```

Comment:
Safe assumption due to data format? Maybe drop an example of what that format is actually parsing in a comment so we can specifically tell.

Reply:
good point